Firestore/Swift/Source/FirestoreDocumentReferable/FirestoreDocumentReferable.swift

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import FirebaseFirestore
+
+public protocol FirestoreDocumentReferable {
+  var documentID: String { get set }
+}
```

Comment:
Thinking some more about this, I'd prefer if we didn't introduce any new mechanisms for expressing a way to map Firestore document IDs to Swift. 

I am aware not everybody is a huge fan of `@DocumentID`, but it is the official mechanism for expressing a mapping between the Firestore document ID and an attribute on a Swift struct, so we should try and use it here as well, if possible.

Since this is being used for object identity checks in for example when removing documents, we might need to introduce a way for users to provide a custom way to check identity if they don't want to map document IDs.